server/render-hydrogen-to-string.js

```diff
@@ -78,21 +78,27 @@ async function renderHydrogenToString({ fromTimestamp, roomData, events, stateEv
     const hydrogenRenderScript = new vm.Script(hydrogenRenderScriptCode, {
       filename: 'hydrogen-vm-render-script.js',
     });
+    console.log('before runInContext');
     const vmResult = hydrogenRenderScript.runInContext(vmContext);
+    console.log('after runInContext');
     // Wait for everything to render
     // (waiting on the promise returned from `hydrogen-render-script.js`)
     await vmResult;
 
     const documentString = dom.document.body.toString();
+    console.log('got documentString');
```

Comment:
Does this even make sense? It somehow was able to return the `documentString` but also throw the error

```sh
$ oc logs mpa-statefulset-0 --previous

> matrix-public-archive@0.0.0 start
> node server/server.js

server process.env.NODE_ENV prod
Config is using nodeEnv=prod
before runInContext
after runInContext
got documentString
/app/node_modules/hydrogen-view-sdk/lib-build/hydrogen.cjs.js:12481
    let stores = [this._storage.storeNames.timelineEvents];
                                ^

TypeError: Cannot read properties of undefined (reading 'storeNames')
    at TimelineReader.readById (/app/node_modules/hydrogen-view-sdk/lib-build/hydrogen.cjs.js:12481:33)
    at Timeline._getEventFromStorage (/app/node_modules/hydrogen-view-sdk/lib-build/hydrogen.cjs.js:12760:46)
    at Timeline._loadContextEntryNotInTimeline (/app/node_modules/hydrogen-view-sdk/lib-build/hydrogen.cjs.js:12745:35)
    at Timeline._loadContextEntriesWhereNeeded (/app/node_modules/hydrogen-view-sdk/lib-build/hydrogen.cjs.js:12739:14)
    at Timeline.addEntries (/app/node_modules/hydrogen-view-sdk/lib-build/hydrogen.cjs.js:12697:10)
    at mountHydrogen (hydrogen-vm-render-script.js:192:12)
    at hydrogen-vm-render-script.js:336:1
    at Script.runInContext (node:vm:139:12)
    at renderHydrogenToString (/app/server/render-hydrogen-to-string.js:82:43)
    at async /app/server/routes/install-routes.js:153:34
```

Reply:
Is the VM continuing to run after returning the result?

Reply:
> Is the VM continuing to run after returning the successful result?

Yes, it's continuing to run. I assumed it closed after the result returned but it runs forever and there isn't a close or stop method so we can do it ourselves.

You can reproduce this by adding some `setTimeout` to `shared/hydrogen-vm-render-script.js`

```js
  setTimeout(() => {
    throw new Error('aewffewaaewfafew');
  }, 5000);

  setInterval(() => {
    console.log('.');
  }, 1000);
```

Reply:
Solved by running all of this inside a `child_process` so we can exit the whole process that does the render after we get the initial SSR HTML.